src/Dns/DotClient.cs

```diff
@@ -0,0 +1,475 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Security;
+using System.Net.Sockets;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using Common.Logging;
+using Nito.AsyncEx;
+
+namespace Makaretu.Dns
+{
+    /// <summary>
+    ///     Client to a DNS server over TLS.
+    /// </summary>
+    /// <remarks>
+    ///     DNS over TLS is a security protocol for encrypting and wrapping
+    ///     DNS queries and answers via the Transport Layer Security (TLS) protocol. The goal
+    ///     is to increase user privacy and security by preventing eavesdropping and
+    ///     manipulation of DNS data via man-in-the-middle attacks.
+    ///     <para>
+    ///         All queries are padded to the closest multiple of <see cref="BlockLength" /> octets.
+    ///     </para>
+    /// </remarks>
+    /// <seealso href="https://tools.ietf.org/html/rfc7858" />
+    /// <seealso href="https://tools.ietf.org/html/rfc8310" />
+    public class DotClient : DnsClientBase
+    {
+        private SslStream dnsServer;
+        private readonly AsyncLock dnsServerLock = new AsyncLock();
+        private readonly Random rng = new Random();
+
+        /// <summary>
+        ///     The default port of a DOT server.
+        /// </summary>
+        public const int DefaultPort = 853;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this constant to a 'static' read-only property.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)